lib/aliases.rb

```diff
@@ -36,9 +36,15 @@ def remove(name)
     end
 
     def each(only)
+      # Only allow these file types
+      allowed_ftypes = ["file", "link"]
+
       Dir["#{BASE_DIR}/*"].each do |path|
         next if path.end_with? "~" # skip Emacs-like backup files
 
+        # Skip non-allowed file types
+        next unless allowed_ftypes.include?(File.ftype(path))
```

Comment:
```suggestion
        # Skip non-allowed file types
        next if File.directory?(path)
```

Reply:
I think it makes sense to skip non-regular files and non-symlinks, rather than just directories: FIFO special, block special, character special, and socket — WDYT?

Reply:
My point is that `brew alias` wants to read only regular files (and symlinks to regular files probably) and thus only allow specific file types explicitly. Obviously in most cases that's not a matter as it is not expected for users to mess with aliases directory, though why don't we implement a more robust (while still simple) solution while we're here instead of remediating the single use case that only I have come across and which wasn't supposed to exist while it is implicitly allowed to exist.
I'm not familiar with Ruby and hence I don't know whether `File.readlines` at line 48 would refuse to read e.g. socket, but if it would read, then it can be used to implement various types of unwanted behavior.
I intentionally am not trying to implement check for "_symlinks to regular files_" as it would indeed complicate this simple function.

Please let me know if, as of above outlined, you still do want me to exclude only directories from the loop (and hence allow any other type of files). Thanks.

Reply:
> you still do want me to exclude only directories from the loop

Yes, please.

Reply:
Done